crypto/s2n_ecdsa.c

```diff
@@ -157,26 +161,26 @@ int s2n_ecdsa_signature_size(const struct s2n_ecdsa_private_key *key)
     return ECDSA_size(key->eckey);
 }
 
-int s2n_ecdsa_keys_match(const struct s2n_ecdsa_public_key *pub_key, const struct s2n_ecdsa_private_key *priv_key) 
+int s2n_ecdsa_keys_match(const struct s2n_ecdsa_public_key *pub_key, struct s2n_hash_state *state_in,
+                         const struct s2n_ecdsa_private_key *priv_key, struct s2n_hash_state *state_out)
```

Comment:
I don't like how this signature is now exposing the internals of how the function works. It feels to me like a slight violation of encapsulation. To me, the caller should just be able to call a function that says, "given this key and this other key, do they match?" It shouldn't need to provide a hash algorithm to be able to help prove it. Is this just to try to avoid having to reallocate hash state?